### PR TITLE
Restart SwSS, syncd and dependent services if a critical process in syncd container exits unexpectedly

### PR DESCRIPTION
**Pull request by jleveque**
**- What I did**

Restart SwSS, syncd and dependent services if a critical process in the syncd container exits unexpectedly

**- How I did it**

Add the same mechanism I developed for the SwSS service in https://github.com/Azure/sonic-buildimage/pull/2845 to the syncd service. However, in order to cause the SwSS service to *also* exit and restart in this situation, I developed a `docker-wait-any` program which the SwSS service uses to wait for *either* the swss *or* syncd containers to exit.

**- How to verify it**

Run etiher `sudo pkill -11 <critical_process_in_syncd_container>`, and observe that syncd service exits, swss and all dependent services exit, then all of those services start back up.

### CODE REVIEW

**Comment by jleveque** [comment]
Retest vsimage please

File: files/scripts/swss.sh [hunk 1]
```diff
@@ -131,7 +131,13 @@ start() {
 
 wait() {
     start_peer_and_dependent_services
-    /usr/bin/${SERVICE}.sh wait
+
+    # Allow peer service some time to start up
+    sleep 5
```
Comment (lguohan) [inline files/scripts/swss.sh]: why is this needed before?

Comment (jleveque) [reply files/scripts/swss.sh]: This is due to the addition of the `docker-wait-any` script. I want to ensure that the syncd container is up and running before starting `docker-wait-any`, because if it isn't running when the script is kicked off, the script will exit immediately.

Comment (lguohan) [reply files/scripts/swss.sh]: it is better to have a loop per second, and check if syncd started, if yes then call docker-wait-any.

Comment (jleveque) [reply files/scripts/swss.sh]: What if for some reason the syncd container fails to start? Should we set a max wait time?

Comment (lguohan) [reply files/scripts/swss.sh]: yes, for example, 60 seconds, which is much safer.

Comment (jleveque) [reply files/scripts/swss.sh]: Done in commit 3a9028ebe5cd7f4fc6334e7a3194b378762f7f6a

**Comment by lguohan** [comment]
retest broadcom please
